### ISSUE TITLE
Can't bind Ctrl-MouseWheel

### ISSUE THREAD

**Issue by nolanl** [jwilm/alacritty#2512]
Linux/X11 w/ i3wm.  Alacritty 0.3.2.

Possibly pilot error, but I am not able to bind Control+Mouse Wheel actions to IncreaseFontSize/DecreaseFontSize.  Here is what I have in my config's mouse_bindings section:
```
  - { mouse: '4', mods: Control, action: IncreaseFontSize }
  - { mouse: '5', mods: Control, action: DecreaseFontSize }
```

Also, while the documentation and the error message suggest that numeric identifiers are valid, the code actually requires them to be strings (as I have them above).  Ideally it would accept either.

**Comment by enizor** [comment]
The mouse bindings currently only works for mouse buttons. Numeric identifiers corresponds to additional buttons on your mouse, not scrolling wheel action.

Accepting integers should be possible to add somewhere around https://github.com/jwilm/alacritty/blob/45565bb9ca9ef33e38cf5e0592d3878e898ed325/alacritty_terminal/src/config/bindings.rs#L643

I'm not sure allowing bindings for scrolling wheel events would be easy (you must consider both "lines scrolled" by a wheel and "pixels scrolled" by a touchpad). It would probably make a mess in the config file for a rather niche usage.

**Comment by chrisduerr** [comment]
As @enzior has already correctly stated, it's currently not possible to do this for the mouse wheel.

However I don't think this would be super complex, since we already treat pixel scrolls by touchpads similarly to integer-based scrolling by collecting these fractions until they are 1 or bigger.

However it would still be some additional complexity, so I'm not 100% sure that this is actually necessary at this point, since key bindings should do the job.

**Comment by nolanl** [comment]
Thanks @enizor & @chrisduerr for the explanations.  I guess my X11 bias is showing, I just assumed wheels were modeled as 2 extra buttons.

I'll try a PoC patch and see how bad the complexity gets.

**Comment by halvabner** [comment]
> since key bindings should do the job.

Is your suggestion that I remap Ctrl+mouse_wheel to another key system-wide? Or is there a way to have it only get remapped for when I use Alacritty?

**Comment by kchibisov** [comment]
the suggestion is to expose it from alacritty.

**Comment by GanerCodes** [comment]
why does everything i want in life have to be a 5yr old github issue 😭

**Comment by axgkl** [comment]
A good windowmanager should be able to provide mappings for mouse wheel actions.  

E.g. on dwm:

```c
void altwheel(const Arg *arg) { 
  if (arg->i > 0) system("/home/gk/bin/hw resizesmaller");
  else system("/home/gk/bin/hw resizelarger");
}
/* mouse. 4 and 5 are wheel up down */
static Button buttons[] = {
 ...
  	{ ClkClientWin,         ControlMask,    Button4,        altwheel, {.i = -1 }},
  	{ ClkClientWin,         ControlMask,    Button5,        altwheel, {.i = +1 }},
 ```

and in that script 

```bash
function resize {
    xdotool keyup ctrl
    xdotool key ctrl+$1
    xdotool keydown ctrl
}
```

with $1 set to 'minus' or 'equal', which are the mappings understood by alacritty.

**Comment by DziubanMaciej** [comment]
It would be wonderful to see this feature implemented.

**Comment by zen2** [comment]
Another vote to be able to ctrl-mousewheel !

**Comment by AsharLohmar** [comment]
Yeah, the `Ctrl`+`=`/`Ctrl`+`+` and `Ctrl`+`-`, work fine, but I have the `Ctrl`+`scroll` "flow" so naturally built-in from browsers or even mintty that it feels strange without it.

**Comment by blueray453** [comment]
![Image](https://github.com/user-attachments/assets/444cc13d-6fdf-413f-8417-1e4674813bdf)

If you are willing to use tmux on top of alacritty, like

```
function tmux-as-default-terminal() {
  if [[ -o interactive ]]
  then
    if [ -z ${TMUX} ]
    then
      tmux
    fi
  fi
}

tmux-as-default-terminal "$@"
```

You can also use bash or fish. I am using zsh.

And use the following keybinding in tmux as mentioned [here](https://unix.stackexchange.com/a/604034)

```
bind-key -T root C-WheelUpPane {
       run-shell "xdotool key equal"
}

bind-key -T root C-WheelDownPane {
       run-shell "xdotool key minus"
}
```

Then it should work.

**Comment by kchibisov** [comment]
Just in case, the bindings were added.

**Comment by blueray453** [comment]
Did you mean:

```
{ mouse = "Forward", mods = "Control", action = "IncreaseFontSize" }
{ mouse = "Back", mods = "Control", action = "DecreaseFontSize" }
```

**Comment by kchibisov** [comment]
But you need latest master.
```
{ mouse = "WheelUp", mods = "Control", action = "IncreaseFontSize" }
{ mouse = "WheelDown", mods = "Control", action = "DecreaseFontSize" }
```
